tests/components/hunterdouglas_powerview/test_config_flow.py

```diff
@@ -350,3 +352,79 @@ async def test_form_unsupported_device(
     assert result3["result"].unique_id == "A1B2C3D4E5G6H7"
 
     assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.usefixtures("mock_hunterdouglas_hub")
+@pytest.mark.parametrize("api_version", [1, 2, 3])
+async def test_migrate_entry(
+    hass: HomeAssistant,
+    api_version: int,
+) -> None:
+    """Test migrate to newest version."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"host": "1.2.3.4"},
+        unique_id=MOCK_MAC,
+        version=1,
+        minor_version=1,
+    )
+
+    assert entry.version == 1
+    assert entry.minor_version == 1
+    assert await entry.async_migrate(hass) is False
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.version == 1
+    assert entry.minor_version == 2
+    assert await entry.async_migrate(hass) is True
+
+
+@pytest.mark.usefixtures("mock_hunterdouglas_hub")
+@pytest.mark.parametrize("api_version", [1, 2, 3])
+async def test_migrate_id(
+    hass: HomeAssistant,
+    api_version: int,
+) -> None:
+    """Test migration to new entity ids."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"host": "1.2.3.4"},
+        unique_id=MOCK_MAC,
+        version=1,
+        minor_version=1,
+    )
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity_registry = er.async_get(hass)
```

Comment:
You can import the entity registry as test fixture

Reply:
@joostlek  this is the only one i dont think ive covered - i wasnt sure what you wanted here as i thought it not being a fixture and instead building dynamically from the json inputs provided in `fixtures\` was better for testing purposes ?

Reply:
Nono, I mean, you can make it a test parameter, home assistant will just fill it so you don't have to do the async_get

Reply:
tests still seem to be passing so i suspec that is what you were after 😄